### ISSUE TITLE
Support: flow-typed install resolve error 

### ISSUE THREAD

**Issue by bassettsj** [flowtype/flow-typed#338]
When running the `flow-typed install -o` it will run into an error from the resovle module:
`UNCAUGHT ERROR: Error: Cannot find module 'babel-runtime' from '/path/to/root-project-dir'`

What might be the best way to debug this issue? Is it coming from the cli or my project?

**Comment by jeffmo** [comment]
This sounds like an issue with the CLI... Can you confirm which version of the CLI you're using?

**Comment by bassettsj** [comment]
Here's the exact version

```
$ flow-typed version
flow-typed@2.0.0-beta.9
```


Here's the stack trace I can get from there:

```
UNCAUGHT ERROR: Error: Cannot find module 'babel-runtime' from '/path/to/project'
    at /flow-typed/node_modules/resolve/lib/async.js:46:17
    at process (/flow-typed/node_modules/resolve/lib/async.js:173:43)
    at ondir (/flow-typed/node_modules/resolve/lib/async.js:188:17)
    at load (/flow-typed/node_modules/resolve/lib/async.js:69:43)
    at onex (/flow-typed/node_modules/resolve/lib/async.js:92:31)
    at /flow-typed/node_modules/resolve/lib/async.js:22:47
    at FSReqWrap.oncomplete (fs.js:123:15)
```

**Comment by jeffmo** [comment]
Huh... So I'm able to confirm that `flow-typed@2.0.0-beta.9` properly depends on `babel-polyfill` (which properly depends on `babel-runtime`).

Another clarifying question: Do you have `flow-typed` installed as a `devDependency` of your project? Or are you exclusively using a globally installed version?

If the former, have you already tried blowing away your `node_modules` and re-`npm install`ing just to be sure things aren't in a funky state?

**Comment by bassettsj** [comment]
Using flow-typed as a global dependency instead of a `devDependency`, I tried blowing away and reinstalling `node_modules` no luck. Same issue.

**Comment by bassettsj** [comment]
I also have tried using it as a devDepenedency without luck.

**Comment by tornewuff** [comment]
I get the same error, but for react-scripts. It looks like it's not an install problem with the CLI, but a problem with what it's doing with my actual project dependencies?

```
wolfpuppy ~/projects/foo$ flow-typed install
• Found flow-bin@v0.32.0 installed. Installing libdefs compatible with this version of Flow...
• Found 9 dependencies in package.json. Searching for libdefs...
• Installing 2 libdefs...
  • flow-typed/npm/react-redux_v4.x.x.js already exists!
    └> Use --overwrite to overwrite the existing libdef.
  • flow-typed/npm/redux_v3.x.x.js already exists!
    └> Use --overwrite to overwrite the existing libdef.
UNCAUGHT ERROR: Error: Cannot find module 'react-scripts' from '/home/torne/projects/foo'
    at /home/torne/.npm-global/lib/node_modules/flow-typed/node_modules/resolve/lib/async.js:46:17
    at process (/home/torne/.npm-global/lib/node_modules/flow-typed/node_modules/resolve/lib/async.js:173:43)
    at ondir (/home/torne/.npm-global/lib/node_modules/flow-typed/node_modules/resolve/lib/async.js:188:17)
    at load (/home/torne/.npm-global/lib/node_modules/flow-typed/node_modules/resolve/lib/async.js:69:43)
    at onex (/home/torne/.npm-global/lib/node_modules/flow-typed/node_modules/resolve/lib/async.js:92:31)
    at /home/torne/.npm-global/lib/node_modules/flow-typed/node_modules/resolve/lib/async.js:22:47
    at FSReqWrap.oncomplete (fs.js:123:15)
```

**Comment by tornewuff** [comment]
My deps are:
```
  "devDependencies": {
    "flow-bin": "^0.32.0",
    "react-scripts": "0.6.1"
  },
  "dependencies": {
    "bootstrap": "^3.3.7",
    "react": "^15.3.2",
    "react-bootstrap": "^0.30.3",
    "react-dom": "^15.3.2",
    "react-redux": "^4.4.5",
    "redux": "^3.6.0",
    "redux-actions": "^0.12.0"
  },
```

**Comment by jeffmo** [comment]
@tornewuff: Are you fulling `npm install`-ed?

**Comment by tornewuff** [comment]
Yes, everything is installed and works fine.

**Comment by jeffmo** [comment]
Huh, I'll see if I can repro with your package.json later tonight.

Thanks for the report!

**Comment by tornewuff** [comment]
Removing react-scripts from devDependencies makes flow-typed work fine.

**Comment by jeffmo** [comment]
Aha, this is happening because `react-script` doesn't have any index.js or entry-point file.
Fixing coming soon...

**Comment by csvan** [comment]
Getting the same problem with Grunt deps:

```
• Found flow-bin@v0.33.0 installed. Installing libdefs compatible with this version of Flow...
• Found 34 dependencies in package.json. Searching for libdefs...
• rebasing flow-typed cache...done.
UNCAUGHT ERROR: Error: Cannot find module 'grunt-concurrent' from '/home/christopher/IdeaProjects/flow-angular'
    at /usr/lib/node_modules/flow-typed/node_modules/resolve/lib/async.js:46:17
    at process (/usr/lib/node_modules/flow-typed/node_modules/resolve/lib/async.js:173:43)
    at ondir (/usr/lib/node_modules/flow-typed/node_modules/resolve/lib/async.js:188:17)
    at load (/usr/lib/node_modules/flow-typed/node_modules/resolve/lib/async.js:69:43)
    at onex (/usr/lib/node_modules/flow-typed/node_modules/resolve/lib/async.js:92:31)
    at /usr/lib/node_modules/flow-typed/node_modules/resolve/lib/async.js:22:47
    at FSReqWrap.oncomplete (fs.js:123:15)
```

My package.json:

```
{
  "name": "yeomangeneratedproject",
  "private": true,
  "devDependencies": {
    "autoprefixer-core": "^5.2.1",
    "flow-bin": "^0.33.0",
    "grunt": "^0.4.5",
    "grunt-angular-templates": "^0.5.7",
    "grunt-concurrent": "^1.0.0",
    "grunt-contrib-clean": "^0.6.0",
    "grunt-contrib-compass": "^1.0.0",
    "grunt-contrib-concat": "^0.5.0",
    "grunt-contrib-connect": "^0.9.0",
    "grunt-contrib-copy": "^0.7.0",
    "grunt-contrib-cssmin": "^0.12.0",
    "grunt-contrib-htmlmin": "^0.4.0",
    "grunt-contrib-imagemin": "^1.0.0",
    "grunt-contrib-jshint": "^0.11.0",
    "grunt-contrib-uglify": "^0.7.0",
    "grunt-contrib-watch": "^0.6.1",
    "grunt-filerev": "^2.1.2",
    "grunt-google-cdn": "^0.4.3",
    "grunt-jscs": "^1.8.0",
    "grunt-newer": "^1.1.0",
    "grunt-ng-annotate": "^0.9.2",
    "grunt-postcss": "^0.5.5",
    "grunt-svgmin": "^2.0.0",
    "grunt-usemin": "^3.0.0",
    "grunt-wiredep": "^2.0.0",
    "jasmine-core": "^2.5.2",
    "jit-grunt": "^0.9.1",
    "jshint-stylish": "^1.0.0",
    "karma": "^1.3.0",
    "karma-jasmine": "^1.0.2",
    "karma-phantomjs-launcher": "^1.0.2",
    "phantomjs-prebuilt": "^2.1.12",
    "time-grunt": "^1.0.0"
  },
  "engines": {
    "node": ">=0.10.0"
  },
  "scripts": {
    "test": "karma start test/karma.conf.js"
  },
  "dependencies": {
    "lodash": "^4.16.2"
  }
}
```

**Comment by jeffmo** [comment]
This should be fixed by #349. I'll try to merge and publish that sometime tomorrow -- sorry for the inconvenience.

In the meantime try just removing the erroring dependency from the package.json temporarily when you run `flow-typed install ` (then put it back after)

**Comment by jeffmo** [comment]
Just published `flow-typed@2.0.0-beta.10` which should fix this.
Let me know if you hit any more issues!

**Comment by cem2ran** [comment]
I'm getting the same error running beta.10 on the following dependency:
`"react-native-linear-gradient": "^1.5.13"`

**Comment by jeffmo** [comment]
Hmm, looks like `react-native-linear-gradient` uses a non-standard index.{ios,android}.js file :/

Ok, I'll see if we can change things up to be more resilient to this. Thanks for the report!

**Comment by jeffmo** [comment]
@cem2ran: Ok, I just published `flow-typed@2.0.0-beta.11` with https://github.com/flowtype/flow-typed/commit/d295e7d150309b8e950ee623ef7fb05df18cc4e3, which should fix this.
Please try it out and let me know if you have any more issues!
